### DISCUSSION TITLE
How to synthesize speech using phonetical notation like IPA?

### DISCUSSION THREAD

**Discussion by bguan** [dlutton/flutter_tts#241] [Q&A]
I am writing an app to generate arbitrary phonetically annotated speech, using something like IPA ([International Phonetic Alphabet](https://en.wikipedia.org/wiki/International_Phonetic_Alphabet)).

I understand flutter_tts just delegates to the underlying platform specific speech synthesis engine. I wonder if the author or the user community knows how to do this. e.g. what language code to use, specific notation such as surrounding text in square brackets...

For example, is [SSML](https://cloud.google.com/text-to-speech/docs/ssml) supported. 

Thanks.

**Comment by echogit** [comment]
Hi, I need to speak each phoneme of a word each at a time, so I did some digging and found that you can send this to the TTS:
```
<speak>
    <phoneme alphabet='x-sampa' >En, Fer, mei, </phoneme>
    <phoneme alphabet='x-sampa' ph='ro' >ro</phoneme>
</speak>
```

This will speak the phonemes of the word "Enfermeiro" (nurse in Porguese). The 'ro' if sent alone will be spoken with a sound of a "HO" in english, and it should be a "ro". This was the only way to achieve that.
